File: Sdl.Tridion.Context/DeviceClaims.cs
using System.Collections.Generic;
using Tridion.ContentDelivery.AmbientData;

namespace Sdl.Tridion.Context
{
    public class DeviceClaims
    {

        public DeviceClaims()
        {
        }

        public int DisplayHeight { get { return AmbientDataContext.CurrentClaimStore.Get<int>(ClaimUris.UriDeviceDisplayHeight); } }
        public int DisplayWidth { get { return AmbientDataContext.CurrentClaimStore.Get<int>(ClaimUris.UriDeviceDisplayWidth); } }
        public List<string> InputDevices { get { return AmbientDataContext.CurrentClaimStore.Get<List<string>>(ClaimUris.UriDeviceInputDevices); } }
        public bool IsMobile { get { return AmbientDataContext.CurrentClaimStore.Get<bool>(ClaimUris.UriMobile); } }
        public string Model { get { return AmbientDataContext.CurrentClaimStore.Get<string>(ClaimUris.UriDeviceModel); } }
        public int PixelDensity { get { return AmbientDataContext.CurrentClaimStore.Get<int>(ClaimUris.UriPixelDensity); } }
        public int PixelRatio { get { return AmbientDataContext.CurrentClaimStore.Get<int>(ClaimUris.UriPixelRatio); } }
        public bool IsRobot { get { return AmbientDataContext.CurrentClaimStore.Get<bool>(ClaimUris.UriRobot); } }
        public bool IsTablet { get { return AmbientDataContext.CurrentClaimStore.Get<bool>(ClaimUris.UriTablet); } }
        public string Variant { get { return AmbientDataContext.CurrentClaimStore.Get<string>(ClaimUris.UriDeviceVariant); } }
        public string Vendor { get { return AmbientDataContext.CurrentClaimStore.Get<string>(ClaimUris.UriDeviceVendor); } }
        public string Version { get { return AmbientDataContext.CurrentClaimStore.Get<string>(ClaimUris.UriDeviceVendor); } }

    }
}

File: com/sdl/tridion/context/DeviceClaims.java
package com.sdl.tridion.context;

import java.net.URI;
import java.util.Map;

public class DeviceClaims extends ContextClaims {
	public DeviceClaims(Map<URI, Object> claims) {
		super(claims);
		tafContext = "taf:claim:context:device:";
	}

	public DeviceClaims() {
		super();
		tafContext = "taf:claim:context:device:";
	}

	public int getDisplayHeight() {
		return getIntValue(ClaimUris.UriDeviceDisplayHeight);
	}

	public int getDisplayWidth() {
		return getIntValue(ClaimUris.UriDeviceDisplayWidth);
	}

	public boolean isMobile() {
		return getBooleanValue(ClaimUris.UriMobile);
	}

	public String getModel() {
		return getStringValue(ClaimUris.UriDeviceModel);
	}

	public int getPixelDensity() {
		return getIntValue(ClaimUris.UriPixelDensity);
	}

	public int getPixelRatio() {
		return getIntValue(ClaimUris.UriPixelRatio);
	}

	public boolean isRobot() {
		return getBooleanValue(ClaimUris.UriRobot);
	}

	public boolean isTablet() {
		return getBooleanValue(ClaimUris.UriTablet);
	}

	public String getVariant() {
		return getStringValue(ClaimUris.UriDeviceVariant);
	}

	public String getVendor() {
		return getStringValue(ClaimUris.UriDeviceVendor);
	}

	public String getVersion() {
		return getStringValue(ClaimUris.UriDeviceVendor);
	}
}
